packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -223,12 +224,73 @@ void updateLocalProperties({String projectPath, BuildInfo buildInfo}) {
     settings.writeContents(localProperties);
 }
 
+void findAndReplaceVersionProperties({BuildInfo buildInfo}) {
+
+  // early return, if nothing has to be changed
+  if (buildInfo?.buildNumber == null && buildInfo?.buildName == null) {
+    return;
+  }
+
+  final File appGradle = fs.file('android/app/build.gradle');
```

Comment:
Is it safe here to assume the CWD is correct?  Is it possible to make this an absolute path?

Reply:
I guess yes. There are other code snippets in `gradle.dart` where they do it the same way.